typescript-deno-plugin/src/index.ts

```diff
@@ -108,7 +108,11 @@ class Plugin implements ts.server.PluginModule {
         const enabled = fileNameArg !== undefined
           ? this.#getSetting(args[fileNameArg] as string, "enable")
           : this.#getGlobalSettings().enable;
-        return enabled ? emptyReturn : target.call(ls, ...args);
+        return enabled
+          // we should return a new array, so if someone modifies a previous on
+          // we create a new one on each call
+          ? Array.isArray(emptyReturn) ? [] : emptyReturn
```

Comment:
Not a big deal because the scope of this function is quite contained, but what might be good to do is to assert that an empty array or `undefined` is provided for this argument in order to enforce the assumptions made here.

Reply:
I don't want to have a function that gets called every time, for performance reasons.  The vast majority of the values are `undefined` and invoking a function call just to get back `undefined` (or even a fresh array instance) seems like a totally unnecessary overhead to me.

As you point out the functionality is very constrained, therefore we should make the solution fit the constraints, versus making it address eventualities that will never occur.  Adding runtime type constraints doesn't make a whole lot of sense for a very specific bit of code that is very internal and doesn't change frequently.

If this were a public API to be consumed and reused in other places, maybe.

Reply:
I looked back at the code and realised there are a few objects being passed.  So I re-worked it so that it takes either a return type value, or a function that returns a return type value and for the the objects, it is now a function that returns them.  Still very specialised, but optimised for the use case.

Reply:
I don’t think we should be concerned about the performance implications of function calls unless it can be shown that removing one has an impact on performance.

I think providing a function or undefined here would have been simpler and less error prone (ex. someone providing an array with elements in the future and not noticing what this code does), but again, not a big deal.